framework/src/mesh/MooseMesh.C

```diff
@@ -2188,11 +2193,10 @@ MooseMesh::getCustomPartitioner() const
   return _custom_partitioner;
 }
 
-
 void
-MooseMesh::setCustomPartitioner(Partitioner * custom_partitioner)
+MooseMesh::setCustomPartitioner(Partitioner * partitioner)
 {
-  _custom_partitioner = custom_partitioner;
+  _custom_partitioner = partitioner->clone().release();
```

Comment:
Should `_custom_partitioner` perhaps also be a `UniquePtr`?  That way we can guarantee it cleans up after itself, and we can always call
```
getMesh().partitioner().reset(_custom_partitioner.release());
```
to hand ownership over to the LibMesh mesh's Partitioner...

As it stands now, I am not sure if `_custom_partitioner` will possibly be leaked if `setCustomPartitioner` is called without a corresponding call to `MooseMesh::init()`... or maybe I missed something.